Cython/Compiler/Parsing.py

```diff
@@ -3543,7 +3543,7 @@ def p_c_class_options(s):
             typeobj_name = p_ident(s)
         elif s.systring == 'check_size':
             s.next()
-            check_size = p_atom(s).value
+            check_size = p_ident(s)
```

Comment:
We should check the valid values right here. No need to wait until code generation.
And if a simple adaptation or mapping of the values is needed, this can be done here, too.

Reply:
done, should it be a function?

Reply:
No, looks good to me as it is.